sql/grant_tables/grant_tables.go

```diff
@@ -108,6 +110,32 @@ func (g *GrantTables) GetUser(user string, host string, roleSearch bool) *User {
 	return userEntry
 }
 
+// GetUserPrivileges returns all privileges for the user within the given context. This includes their native privileges
+// along with those granted by any active roles.
+func (g *GrantTables) GetUserPrivileges(ctx *sql.Context) (*User, UserGlobalStaticPrivileges) {
+	client := ctx.Session.Client()
+	user := g.GetUser(client.User, client.Address, false)
+	if user == nil {
+		return nil, NewUserGlobalStaticPrivileges()
+	}
+	userPrivSet := NewUserGlobalStaticPrivileges()
```

Comment:
I could change it to such, but wanted to reduce the total number of calls made.

For a more complex query I could load all of the privileges once and just search the map, rather than one call for global, one call for database, two calls for tables, and three calls for columns. Could perhaps figure out some way to pass all of that to a single function, but not too sure what kind of signature that would be.

I’ll think on it and see if I can come up with something. Perhaps there’s a special struct I can nest this stuff in and pass that as a parameter.